File: POS/src/types/appState.ts
/** Mirrors the Rust AppState struct returned by get_app_state(). */
export interface AppState {
  // KDS
  kds_ws_url: string | null;
  kds_terminal_id: string | null;
  kds_display_settings: string | null;
  kds_groups: string | null;
  // Queue Display
  queue_ws_url: string | null;
  queue_terminal_id: string | null;
  // Customer Display
  cd_ws_url: string | null;
  cd_terminal_id: string | null;
  cd_settings: string | null;
  // Kiosk
  kiosk_pos_url: string | null;
  kiosk_terminal_id: string | null;
  kiosk_position: string | null;
  // Device
  device_role: string | null;
}

File: POS/src-tauri/src/db/models/app_state.rs
#[derive(Debug, serde::Serialize)]
pub struct AppState {
    // KDS
    pub kds_ws_url: Option<String>,
    pub kds_terminal_id: Option<String>,
    pub kds_display_settings: Option<String>,
    pub kds_groups: Option<String>,
    // Queue Display
    pub queue_ws_url: Option<String>,
    pub queue_terminal_id: Option<String>,
    // Customer Display
    pub cd_ws_url: Option<String>,
    pub cd_terminal_id: Option<String>,
    pub cd_settings: Option<String>,
    // Kiosk
    pub kiosk_pos_url: Option<String>,
    pub kiosk_terminal_id: Option<String>,
    pub kiosk_position: Option<String>,
    // Device
    pub device_role: Option<String>,
}
